File: Napakalakirb/lib/cultist_player.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class CultistPlayer < Player
  attr_reader :totalCultistPlayer
  
  @@totalCultistPlayer = 0
  
  def initialize( p , c )
    super(p.name)
    newCopia(p)
    @myCultistCard = c
    @@totalCultistPlayer += 1
  end
  def getCombatLevel
    return super() + super()*70/100 + @myCultistCard.gainedLevels*@@totalCultistPlayer
  end
  def getOponentLevel(m)
    return m.getLevelChangeAgainstCultistPlayer
  end
  def shouldConvert
    return false
  end
  protected :getOponentLevel , :shouldConvert
  
  def giveMeATreasure
    number = rand(visibleTreasures.length)
    return visibleTreasures[number]
  end
  def canYouGiveMeATreasure
    return !visibleTreasures.empty?
  end
  private :giveMeATreasure , :canYouGiveMeATreasure
  protected :getCombatLevel , :getOponentLevel , :shouldConvert
end

File: Napakalaki/src/napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.Random;

/**
 *
 * @author jose
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayer = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p , Cultist c){
        super(p);
        myCultistCard = c;
        totalCultistPlayer++;
    }
    @Override
    protected int getCombatLevel(){
        return super.getCombatLevel() + super.getCombatLevel()*20/100 + myCultistCard.getGainedLevels()*totalCultistPlayer;
    }
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    @Override
    protected boolean shouldConvert(){
        return false;
    }

    private Treasure giveMeATreasure(){
        Random  rnd = new Random();
        int number = (int) (rnd.nextDouble() * super.getVisibleTreasures().size());
        return super.getVisibleTreasures().get(number);
    }
   
    private boolean canYouGiveMeATreasure(){
        return !super.getVisibleTreasures().isEmpty();
    }
    
    public static int getTotalCultistPlayer(){
        return totalCultistPlayer;
    }
    
}
